TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/ScoringAlgorithm.java

```diff
@@ -0,0 +1,92 @@
+package org.firstinspires.ftc.teamcode.auton;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.Servo;
+
+import org.firstinspires.ftc.teamcode.subsystems.hardware.Hardware;
+import org.firstinspires.ftc.teamcode.subsystems.hardware.HardwareElement;
+
+@Autonomous(name = "Scoring Algorithm", group = "auton")
+public class ScoringAlgorithm extends OpMode {
+
+    private Hardware robot;
+    // TODO: Get values for variables.
+    boolean pixelsInClaw = false;
+    double armTargetPos = 0.0;
+    double viperSlideTargetPos = 0.0;
+    double jointTargetPos = 0.0;
+
+    public void init() {
+        telemetry.addData("Status", "Initializing");
+
+        robot = new Hardware(hardwareMap, telemetry);
+        // Introducing servos.
+        robot.introduce(new HardwareElement<>(Servo.class, hardwareMap, "clawServo"));
+        robot.introduce(new HardwareElement<>(Servo.class, hardwareMap, "jointServo"));
+        // Introducing motors and encoders.
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "armMotor"));
+        robot.<DcMotor>get("armMotor").setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightViperSlideMotor"));
+        robot.<DcMotor>get("rightViperSlideMotor").setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "leftViperSlideMotor"));
+        robot.<DcMotor>get("leftViperSlideMotor").setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        telemetry.addData("Status", "Initialized");
+    }
+
+    private void scoring() {
+        telemetry.addData("Scoring Status", "Started");
+
+        setViperSlidePos();
+
+        setArmPos();
+
+        setJointPos();
+
+        while (pixelsInClaw) {
+            robot.<Servo>get("clawServo").setPosition(robot.<Servo>get("clawServo").getPosition() - 0.125);
+            wait();
+        }
+
+        telemetry.addData("Scoring Status", "Scored");
+    }
+
+    public void setArmPos() {
+        telemetry.addData("Scoring Status", "Setting Arm Pos");
+        // Setting ArmPos based on Target.
+        while (robot.<DcMotor>get("armMotor").getCurrentPosition() != armTargetPos) {
```

Comment:
Are we sure `.getCurrentPosition()` will work here? There's a chance the encoder values will reset every time we turn the robot on, so we might need to find a way to "calibrate" them. Not sure.
Unless we were planning to make sure that whenever the robot turned on it was always in a certain spot, which would probably work.